libminifi/test/unit/JsonFlowSerializerTests.cpp

```diff
@@ -731,19 +732,173 @@ TEST_CASE("The encrypted flow configuration cannot be decrypted with an incorrec
   core::flow::AdaptiveConfiguration json_configuration_before{configuration_context};
 
   const auto schema = core::flow::FlowSchema::getNiFiFlowJson();
-  const auto config_json_with_nifi_schema = utils::string::join_pack(config_json_with_nifi_schema_part_1, config_json_with_nifi_schema_part_2);
+  const auto config_json_with_nifi_schema = minifi::utils::string::join_pack(config_json_with_nifi_schema_part_1, config_json_with_nifi_schema_part_2);
   const auto process_group_before = json_configuration_before.getRootFromPayload(config_json_with_nifi_schema);
   REQUIRE(process_group_before);
 
   rapidjson::Document doc;
   rapidjson::ParseResult res = doc.Parse(config_json_with_nifi_schema.data(), config_json_with_nifi_schema.size());
   REQUIRE(res);
   const auto flow_serializer = core::json::JsonFlowSerializer{std::move(doc)};
-  std::string config_json_encrypted = flow_serializer.serialize(*process_group_before, schema, encryption_provider, {});
+  std::string config_json_encrypted = flow_serializer.serialize(*process_group_before, schema, encryption_provider, {}, {});
 
-  const utils::crypto::Bytes different_secret_key = utils::string::from_hex("ea55b7d0edc22280c9547e4d89712b3fae74f96d82f240a004fb9fbd0640eec7");
-  configuration_context.sensitive_values_encryptor = utils::crypto::EncryptionProvider{different_secret_key};
+  const minifi::utils::crypto::Bytes different_secret_key = minifi::utils::string::from_hex("ea55b7d0edc22280c9547e4d89712b3fae74f96d82f240a004fb9fbd0640eec7");
+  configuration_context.sensitive_values_encryptor = minifi::utils::crypto::EncryptionProvider{different_secret_key};
 
   core::flow::AdaptiveConfiguration json_configuration_after{configuration_context};
-  REQUIRE_THROWS_AS(json_configuration_after.getRootFromPayload(config_json_encrypted), utils::crypto::EncryptionError);
+  REQUIRE_THROWS_AS(json_configuration_after.getRootFromPayload(config_json_encrypted), minifi::utils::crypto::EncryptionError);
 }
+
+TEST_CASE("Parameter provider generated parameter context is serialized correctly") {
```

Comment:
These two test cases are identical except for the value of `config_json_with_nifi_schema`, aren't they? If they are, then I think it would be better to merge them into one. This could be done with `GENERATE` or `SECTION` or an ad-hoc helper function.

(Same comment about the YAML tests.)

Reply:
I would keep them separate, as the second test checks that the old parameter value was not overriden, while the first test checks for the newly set test value gathered from the environment variable.

Reply:
OK, some code duplication is acceptable in unit tests if it makes the logic clearer.